addons/pos_restaurant/static/src/overrides/models/pos_store.js

```diff
@@ -311,9 +311,9 @@ patch(PosStore.prototype, {
     getTableOrders(tableId) {
         return this.get_open_orders().filter((order) => order.table_id?.id === tableId);
     },
-    async unsetTable() {
+    unsetTable() {
         try {
-            await this.syncAllOrders();
+            this.syncAllOrders();
```

Comment:
Restore this diff. we still need the procedures under this method to be synchronous. I think calling `unsetTable` without `await` should be enough to satisfy the spec. Also, you can just put the call in `onMounted`. You can remove the `onWillStart`.